### ISSUE TITLE
Cannot reuse identities in auth0-dev

### ISSUE THREAD

**Issue by viorelaioia** [mozilla-iam/auth0-deploy#182]
**STR:**
1. Have a mozillians account with a github 2fa and a passwordless identity (in mozillians staging)
2. Delete the github identity from mozillians staging admin
3. Delete the github identity from auth0-dev
4. Logout from mozillians
5. Try to login passwordless, as the github identity was deleted from both auth0-dev and mozillians staging

**Expected:**
Passwordless login is successful.

**Actual:**
"Oye, something wrong" error message is shown in sso forbidden page. 

**Notes:**
I deleted a mozillians account that uses Mozilla LDAP login identity (iam_test@mozilla.com) from mozillians staging, in order to use it in an upgrade account scenario. I tried to verify that LDAP identity and I received an error in mozillians. Afterwards, I tried to login to mozillians again with that LDAP identity and I got the registration page.

**This is a blocker in testing the nlx changes that should go in the next release, as I cannot reuse the identities that I already have** (especially the LDAP identities, I cannot test "upgrade from github to LDAP scenarios").

**Comment by viorelaioia** [comment]
CC @andrewkrug , @gdestuynder

**Comment by andrewkrug** [comment]
@viorelaioia does the management API client that we enabled to delete users handle this problem?

**Comment by viorelaioia** [comment]
@andrewkrug , I'm using that client in order to delete the identity, the deletion seems to be successful, but I still cannot reuse the identity.

**Comment by andrewkrug** [comment]
@gdestuynder do we cache those anywhere to determine the account upgrade scenarios on the auth0 tenant?

**Comment by gdestuynder** [comment]
there's no caching of identities, once deleted its gone

**Comment by andrewkrug** [comment]
So the only other reason might be in mozillians ... some log or database?

**Comment by gdestuynder** [comment]
![image](https://user-images.githubusercontent.com/956866/37547418-c52b7ecc-292e-11e8-82df-48a16e9298d1.png)
this is what we have on dev atm

**Comment by akatsoulas** [comment]
@viorelaioia are you trying to login with passwordless with the iam_test identity?

**Comment by viorelaioia** [comment]
@akatsoulas , no. I was trying to upgrade my account from github to LDAP.

**Comment by viorelaioia** [comment]
After more debugging with @andrewkrug , looks like I had two github accounts associated with the same primaryEmail and I was getting that error because I didn't delete the correct github identity.
After deleting the correct identity, I managed to login passwordless and try the upgrade scenario. 

As for the LDAP identity, after getting the "Account verification failed" error message (when trying to upgrade from github to ldap), I went to auth0-dev and noticed that the LDAP identity was still existing - probably it was somehow re-created when trying to add it to the github account - so I deleted it again. Afterwards, I managed to upgrade successfully. 

That being said, I am not blocked anymore on any of these accounts, so I'm closing this issue.
